install/infra/modules/eks/kubernetes.tf

```diff
@@ -1,24 +1,65 @@
+locals {
+  private_primary_subnet_cidr   = cidrsubnet(var.vpc_cidr, 7, 0)
+  private_secondary_subnet_cidr = cidrsubnet(var.vpc_cidr, 7, 1)
+  public_primary_subnet_cidr    = cidrsubnet(var.vpc_cidr, 7, 2)
+  public_secondary_subnet_cidr  = cidrsubnet(var.vpc_cidr, 7, 3)
+  public_db_subnet_cidr_1       = cidrsubnet(var.vpc_cidr, 7, 4)
+  public_db_subnet_cidr_2       = cidrsubnet(var.vpc_cidr, 7, 5)
+}
+
 module "vpc" {
   source  = "terraform-aws-modules/vpc/aws"
   version = "3.12.0"
 
   name                 = "vpc-${var.cluster_name}"
   cidr                 = var.vpc_cidr
   azs                  = var.vpc_availability_zones
-  private_subnets      = [var.private_primary_subnet_cidr, var.private_secondary_subnet_cidr]
-  public_subnets       = [var.public_primary_subnet_cidr, var.public_secondary_subnet_cidr, var.public_db_subnet_cidr_1, var.public_db_subnet_cidr_2]
+  private_subnets      = [local.private_primary_subnet_cidr, local.private_secondary_subnet_cidr]
+  public_subnets       = [local.public_primary_subnet_cidr, local.public_secondary_subnet_cidr, local.public_db_subnet_cidr_1, local.public_db_subnet_cidr_2]
   enable_nat_gateway   = true
   enable_dns_hostnames = true
 }
 
+resource "aws_security_group_rule" "eks-worker-ingress-self" {
+  description              = "Allow node to communicate with each other"
+  from_port                = 0
+  protocol                 = "-1"
+  security_group_id        = aws_security_group.nodes.id
+  source_security_group_id = aws_security_group.nodes.id
+  to_port                  = 65535
+  type                     = "ingress"
+}
+
+resource "aws_security_group_rule" "eks-worker-ingress-cluster" {
+  description              = "Allow worker Kubelets and pods to receive communication from the cluster control plane"
+  from_port                = 1025
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.nodes.id
+  source_security_group_id = aws_security_group.nodes.id
+  to_port                  = 65535
+  type                     = "ingress"
+}
+
+### Worker Node Access to EKS Master
+resource "aws_security_group_rule" "eks-cluster-ingress-node-https" {
+  description              = "Allow pods to communicate with the cluster API Server"
+  from_port                = 443
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.nodes.id
+  source_security_group_id = aws_security_group.nodes.id
+  to_port                  = 443
+  type                     = "ingress"
+}
+
+
 resource "aws_security_group" "nodes" {
   name   = "nodes-sg-${var.cluster_name}"
   vpc_id = module.vpc.vpc_id
 
   ingress {
```

Comment:
As mentioned earlier, are there any interactions of this inline rule and the separate `aws_security_group_rule` resources?

Reply:
removed the inline `ingress` rule.